content/latest/api/ycql/type_jsonb.md

```diff
@@ -197,6 +197,38 @@ cqlsh> SELECT * FROM store.books WHERE id = 4;
   4 | {"author":{"first_name":"John","last_name":"Doe"},"editors":["Robert","Jack","Melisa"],"genre":"novel","name":"Great Expectations","year":1950}
 ```
 
+- Update a missing JSONB subdocument resulting in an insert.
+
+```sql
+cqlsh> UPDATE store.books SET details->'publisher' = '"Chapman and Hall"' WHERE id = 4;
+```
+
+```sql
+cqlsh> SELECT * FROM store.books WHERE id = 4;
+```
+
+```
+ id | details
+----+-------------------------------------------------------------------------------------------------------------------------------------------------
+  4 | {"author":{"first_name":"John","last_name":"Doe"},"editors":["Robert","Jack","Melisa"],"genre":"novel","name":"Great Expectations","publisher":"Chapman and Hall","year":1950}
+```
+
+- Update a multi-hop missing JSONB subdocument resulting in an error.
```

Comment:
NOTE: I am unsure about whether this second example is needed.

Reply:
Instead of adding this as another section, we should just add a note saying something like
"Note that JSONB upsert only works for JSON objects and not for other data types like arrays, integers, strings etc. Additionally, only the leaf property of an object will be inserted if it is missing. We do not support upsert on non-leaf properties."
Then, add two examples:
1. Where you try to run upsert on json array
2. Your current example.